src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MemberResolutionKind.cs

```diff
@@ -115,6 +115,12 @@ internal enum MemberResolutionKind : byte
         /// </summary>
         StaticInstanceMismatch,
 
+        /// <summary>
+        /// The candidate member was rejected because its calling convention did not match the function pointer
+        /// calling convention.
+        /// </summary>
+        WrongCallingConvention,
```

Comment:
>WrongCallingConvention [](start = 8, length = 22)

I believe we have a function that reports diagnostics for candidates in priority order (there are better candidates among bad). I would expect that function adjusted to check for this failure kind in the right order and report specific error.  Making it very clear that the reason is a calling convention mismatch.